service/waterfall.go

```diff
@@ -546,51 +668,63 @@ func (uis *UIServer) waterfallPage(w http.ResponseWriter, r *http.Request) {
 		skip = 0
 	}
 
-	finalData := waterfallData{}
+	variantQuery := strings.TrimSpace(r.URL.Query().Get(BVFilterParam))
 
 	// first, get all of the versions and variants we will need
-	vvData, err := getVersionsAndVariants(skip,
-		VersionItemsToCreate, project)
+	vvData, err := getVersionsAndVariants(
+		skip, VersionItemsToCreate, project, variantQuery,
+	)
+
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusNotFound, err)
+		return
+	}
 
+	finalData, err := waterfallDataAdaptor(vvData, project, skip, variantQuery)
 	if err != nil {
 		uis.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
-	var wfv waterfallVersions = vvData.Versions
 
-	sort.Sort(wfv)
-	finalData.Versions = wfv
+	uis.WriteHTML(w, http.StatusOK, struct {
+		Data     waterfallData
+		JiraHost string
+		ViewData
+	}{finalData, uis.Settings.Jira.Host, uis.GetCommonViewData(w, r, false, true)}, "base", "waterfall.html", "base_angular.html", "menu.html")
+}
 
-	sort.Sort(vvData.BuildVariants)
-	rows := []waterfallRow{}
-	for _, bv := range vvData.BuildVariants {
-		rows = append(rows, vvData.Rows[bv.Id])
+func (restapi restAPI) getWaterfallData(w http.ResponseWriter, r *http.Request) {
```

Comment:
This should be a restV2 route, and ideally should be the existing GetVersionsAndVariants route. Do you think it'll be a lot of effort to have both this and the one that perf discovery uses be the same route? If so could you open an EVG ticket to combine them?